core/src/main/java/com/scalar/db/io/BlobValue.java

```diff
@@ -110,29 +112,35 @@ public boolean equals(Object o) {
       return false;
     }
     BlobValue other = (BlobValue) o;
-    return this.name.equals(other.name) && bytes.equals(other.bytes);
+    if (!name.equals(other.name)) {
+      return false;
+    }
+    if (value.isPresent() && other.value.isPresent()) {
+      return Arrays.equals(value.get(), other.value.get());
+    }
+    return !value.isPresent() && !other.value.isPresent();
   }
 
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
         .add("name", name)
-        .add("value", bytes.toString())
+        .add("value", value.map(Arrays::toString).toString())
```

Comment:
```suggestion
        .add("value", value.map(Arrays::toString).get())
```

Reply:
I think `get()` throws `NoSuchElementException` if value is empty. So should we keep `toString()`?

Reply:
Ah, OK. Then, let's keep it.